### PR TITLE
ENH: helpful git shortcuts

### PR DESCRIPTION
**Pull request by ZLLentz**
git_setup_fork -> clone my repo, add the upstream as a remote
git_sync_upstream -> pull the upstream, push to my remote

configurable via environment variables

### CODE REVIEW

File: on_site/bash_functions [hunk 1]
```diff
@@ -26,6 +26,47 @@ sshcd() {
...
+  usage="git_setup_fork org/repo"
+  if [ -z "${1}" ]; then
+    echo "${usage}"
+    return
+  fi
+  REPO=`basename "${1}"`
+  UPSTREAM=`dirname "${1}"`
+  git clone --origin "${GITHUB_ORIGIN:-origin}" "git@github.com:${GITHUB_USERNAME:-${USER}}/${REPO}.git"
+  pushd "${REPO}"
+  git remote add "${GITHUB_UPSTREAM:-upstream}" "git@github.com:${UPSTREAM}/${REPO}.git"
```
Comment (klauer) [inline on_site/bash_functions:51]: I think we can avoid `pushd`/`popd` with the following:
```suggestion
  git --git-dir "$REPO" remote add "${GITHUB_UPSTREAM:-upstream}" "git@github.com:${UPSTREAM}/${REPO}.git"
```

**Review by klauer — APPROVED** [review]
This is nice! I'd like to add some help information in `helpme.md` about git.
I'd like to have this in there along with the handy `gh` CLI tool we have in our conda repo.

**Comment by ZLLentz** [comment]
Also, I should copy this to `off_site/bash_functions` file too, right? Or is this in the wrong file altogether for something that isn't necessarily on-site/off-site specific?

**Comment by klauer** [comment]
> Also, I should copy this to off_site/bash_functions file too, right? Or is this in the wrong file altogether for something that isn't necessarily on-site/off-site specific?

It's something I didn't come up with a good pattern for just yet. For each location, I was hoping for just _one_ obvious file you could go to look for your aliases/functions/etc - which if we start branching out into separate files could obfuscate things a bit.

For now, I think copying would be an easy way to avoid making a decision on it now. Perhaps the future includes things like: `shared/bashrc_git` or something like that.

File: on_site/bash_functions (hunk 1 above)
Comment (ZLLentz) [reply on_site/bash_functions:51]: Some small details regarding --git-dir and -C args, I think -C is actually the correct one here (my hint was getting an error when I used --git-dir)
https://git-scm.com/docs/git#Documentation/git.txt--Cltpathgt
https://git-scm.com/docs/git#Documentation/git.txt---git-dirltpathgt

**Comment by ZLLentz** [comment]
Actually I think I'll start a git section in the helpme before merging- at least with some instructions about modifying to set your github username

**Review by klauer — APPROVED** [review]
Documentation additions are fantastic! Merging

File: on_site/bash_functions (hunk 1 above)
Comment (klauer) [reply on_site/bash_functions]: I see now - good to know 👍

**Pull request merged**